src/System.Runtime/tests/System/LazyTests.cs

```diff
@@ -103,17 +106,283 @@ public static void ToString_DoesntForceAllocation()
+            }
+        }
+
+        private static void DifferentException<T>(Lazy<T> x)
+        {
+            Exception first = null;
+            try
+            {
+                var _ = x.Value;
+            }
+            catch (Exception thrown1)
+            {
+                first = thrown1;
+            }
+            Assert.NotNull(first);
+
+            Exception second = null;
+            try
+            {
+                var _ = x.Value;
+            }
+            catch (Exception thrown2)
+            {
+                second = thrown2;
+            }
+            Assert.NotNull(second);
+
+            Assert.NotEqual(first, second);
+        }
+
+        [Fact]
+        public static void SameOrDifferentException()
+        {
+            SameException(new Lazy<int>(() => { throw new MyException(99); }));
+            SameException(new Lazy<int>(() => { throw new MyException(99); }, true));
+            SameException(new Lazy<int>(() => { throw new MyException(99); }, false));
+            SameException(new Lazy<int>(() => { throw new MyException(99); }, LazyThreadSafetyMode.ExecutionAndPublication));
+            SameException(new Lazy<int>(() => { throw new MyException(99); }, LazyThreadSafetyMode.None));
+
+            DifferentException(new Lazy<int>(() => { throw new MyException(99); }, LazyThreadSafetyMode.PublicationOnly));
+
+            DifferentException(new Lazy<ExceptionInCtor>());
+            DifferentException(new Lazy<ExceptionInCtor>(true));
+            DifferentException(new Lazy<ExceptionInCtor>(false));
+            DifferentException(new Lazy<ExceptionInCtor>(LazyThreadSafetyMode.ExecutionAndPublication));
+            DifferentException(new Lazy<ExceptionInCtor>(LazyThreadSafetyMode.None));
+
+            DifferentException(new Lazy<ExceptionInCtor>(LazyThreadSafetyMode.PublicationOnly));
+
+            SameException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99)));
+            SameException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99), true));
+            SameException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99), false));
+            SameException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99), LazyThreadSafetyMode.ExecutionAndPublication));
+            SameException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99), LazyThreadSafetyMode.None));
+
+            DifferentException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99), LazyThreadSafetyMode.PublicationOnly));
+        }
+
+        [Fact]
+        static void Serialization_ValueType()
+        {
+            var stream = new MemoryStream();
+            var formatter = new BinaryFormatter();
+            formatter.Serialize(stream, new Lazy<int>(() => 42));
+            stream.Seek(0, SeekOrigin.Begin);
+
+            var fortytwo = (Lazy<int>)formatter.Deserialize(stream);
+            Assert.True(fortytwo.IsValueCreated);
+            Assert.Equal(fortytwo.Value, 42);
+        }
```

Comment:
Argh, running out of time and for some reason this just isn't working... I can see it used in `ModuleTests.cs` and `PointerTests.cs`, and I can see in the project file that there is a link to `BinaryFormatterHelper.cs`, but for some reason it just isn't being picked up; I'm just not seeing the `System.Runtime.Serialization.Formatters.Tests` namespace at all, and my time is running out, so I've left this as it originally was (baring the `public` change)